assets/dev/js/editor/elements/views/widget.js

```diff
@@ -129,18 +129,24 @@ const WidgetView = BaseWidget.extend( {
 		// TODO: Find a better way to detect if all the images have been loaded
 		self.$el.imagesLoaded().always( function() {
 			setTimeout( function() {
-				// Since 'outerHeight' will not handle hidden elements, and mark them as empty (e.g. nested tabs).
-				const $widgetContainer = self.$el.children( '.elementor-widget-container' ).length ? self.$el.children( '.elementor-widget-container' ) : self.$el,
```

Comment:
```
const $widgetContainer = self.$el.children( '.elementor-widget-container' ).length ? self.$el.children( '.elementor-widget-container' ) : self.getDomElement(),
```

I think this one will solve all the issues, without the need for the packages code

Reply:
I also think you have to change the `handleEmptyWidget`
something like that:
```
this.$el.addClass( 'elementor-widget-empty' );

		const icon = '<i class="elementor-widget-empty-icon ' + this.getEditModel().getIcon() + '"></i>';

		if ( this.$el === this.getDomElement() ) {
		    this.$el.append( icon );
		} else {
			this.$el.html( icon );
		}
```

Reply:
Thank you, Nevo. I wasn't aware of the `getDomElement()` method.
Both implemented.

Reply:
I have reverted most of these changes.